src/KMonad/App/Parser/Keycode.hs

```diff
@@ -0,0 +1,118 @@
+module KMonad.App.Parser.Keycode
+  ( -- * $parser
+    keycodeP
+
+    -- * $aliases
+  , keycodeAliases
+  )
+where
+
+import KMonad.Prelude
+import KMonad.Util.Keyboard
+import KMonad.Util.Name
+import KMonad.App.Parser.Operations
+import KMonad.App.Parser.Types
+
+import qualified RIO.HashMap as M
+
+--------------------------------------------------------------------------------
+-- $parser
+--
+-- The core parser that reads keycodes from raw-text. This is defined as a large
+-- hashmap that is constructed by adding a bunch of aliases on top of the
+-- collection of core names.
+--
+-- NOTE: These aliases only exist within the scope of parsing. Once the config
+-- has been parsed, we only refer to 'Keycode's by their 'CoreName'. In the
+-- future we might add support for locales, at which point we will add that
+-- remapping separately.
+
+-- | Parse a keycode
+keycodeP :: Parser Keycode
+keycodeP = fromLexicon allNames <?> "keycode" where
+  -- Lexicon of keycode names including aliases
+  allNames  = case foldM (flip insertAlias) coreNames keycodeAliases of
+    Left err  -> throw err
+    Right lex -> lex
+  -- Lexicon of core keycode names as text
+  coreNames = M.fromList . map (first unCore) . M.toList $ keycodeNames
+
+-- | A collection of alternative names for existing 'CoreName's
+--
+-- NOTE: Our unit tests guarantee that:
+-- * This list only provides aliases to valid 'CoreName's
+-- * No name ever gets overwritten
+keycodeAliases :: [Alias]
+keycodeAliases = foldMap (\(c, as) -> map (,unCore c) as)
```

Comment:
We're missing `pause` as an aliase for `paus` here

Reply:
Well caught! It's all these little things that are so hard to catch on your own. Much appreciated.